WasatchNET/AndorSpectrometer.cs

```diff
@@ -207,28 +213,84 @@ protected override double[] getSpectrumRaw(bool skipTrigger = false)
+                uint success = andorDriver.GetAcquiredData(spec, (uint)(pixels));
+
+                if (success != DRV_SUCCESS)
+                    return null;
+
+                double[] convertedSpec = Array.ConvertAll(spec, item => (double)item);
+
+
+                if (eeprom.featureMask.invertXAxis)
+                    Array.Reverse(convertedSpec);
 
-            if (success != DRV_SUCCESS)
-                return null;
+                logger.debug("getSpectrumRaw: returning {0} pixels", spec.Length);
+                return convertedSpec;
+            }
+            else
+            {
+                int[] spec = new int[yPixels * pixels / BINNING];
 
-            double[] convertedSpec = Array.ConvertAll(spec, item => (double)item);
+                // ask for spectrum then collect, NOT multithreaded (though we should look into that!), blocks
+                spec = new int[yPixels * pixels / BINNING];     //defaults to all zeros
+                andorDriver.StartAcquisition();
+                andorDriver.WaitForAcquisition();
+                uint success = andorDriver.GetAcquiredData(spec, (uint)(yPixels * pixels / BINNING));
+
+                if (success != DRV_SUCCESS)
+                    return null;
 
+                double[] convertedSpec = Array.ConvertAll(spec, item => (double)item);
 
-            if (eeprom.featureMask.invertXAxis)
-                Array.Reverse(convertedSpec);
+                if (eeprom.featureMask.invertXAxis)
+                    Array.Reverse(convertedSpec);
+
+                logger.debug("getSpectrumRaw: returning {0} pixels", spec.Length);
+                return convertedSpec;
+            }
 
-            logger.debug("getSpectrumRaw: returning {0} pixels", spec.Length);
-            return convertedSpec;
         }
 
 
+        public override bool areaScanEnabled
+        {
+            get
+            {
+                return areaScanEnabled_;
+            }
+            set
+            {
+                areaScanEnabled_ = value;
+                lock (acquisitionLock)
+                {
+                    if (value)
+                    {
+                        uint errorValue = andorDriver.SetReadMode(4);
+                        errorValue = andorDriver.SetImage(1, BINNING, 1, (int)pixels, 1, yPixels);
+                        if (errorValue != DRV_SUCCESS)
+                            areaScanEnabled_ = false;
+                        andorDriver.SetShutterEx(1, 1, SHUTTER_SPEED_MS, SHUTTER_SPEED_MS, 0);
+
+                    }
+                    else
+                    {
+                        // Set readout mode to full vertical binning
+                        uint errorValue = andorDriver.SetReadMode(0);
+                        andorDriver.SetShutterEx(1, 1, SHUTTER_SPEED_MS, SHUTTER_SPEED_MS, 0);
+                    }
+                }
+            }
+        }
+
         public override bool highGainModeEnabled
```

Comment:
My understanding is the iDus does have this, and we should eventually support it (not necessarily today though).